docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,827 @@
+does not declare the parent-child relation in the topics either, as follows:
+
+```
+tedge/<entity-id>/...
+```
+
+...where `id` could be the `id` of the main device, child device or a service on any of them.
+The alias `main` can still be used for the main tedge device.
+
+Examples:
+* Main device measurements: `tedge/<tedge-device-id>/telemetry/measurements`
+* Main device service measurements: `tedge/<service-id>/telemetry/measurements`
+* Child device measurements: `tedge/<child-device-id>/telemetry/measurements`
+* Child device service measurements: `tedge/<child-service-id>/telemetry/measurements`
+
+```admonish note
+`tedge/main` could be just used as an alias for `tedge/<tedge-device-id>` for simplicity.
+```
+
+**Pros**
+
+1. Due to the lack of distinction in the topics between parent devices and child devices,
+   it is easier to write code for "the device" irrespective of whether it is deployed as a parent or child.
+1. Minimal payload size as we're only including the `id`s everywhere without repeating any context info.
+
+**Cons**
+
+1. Not easy to differentiate the context(from parent or child) easily from the message.
+1. The `id`s must be unique between all devices and services in a deployment.
+1. Not easy to do subscriptions like: "measurements only from child devices or only from services, excluding parent",
+   without keeping track of all the `id`s of services registered with that child device.
+   Easy wild card subscriptions are not possible at all.
+1. No automatic registration as bootstrapping is mandatory for everything.
+
+## Proposal 3: Namespace extensible topics
+
+This scheme splits the topic levels into two parts:
+* entity/component identification levels
+* data type identification levels
+
+This scheme reserves 4 subtopic levels for entity identification,
+and 2 levels for data type identification such as measurements, alarms, events, commands etc as follows:
+
+```
+te/<name-space-1>/<name-space-2>/<name-space-3>/<name-space-4>/<data-type>/<data-instance-type>
+```
+
+Having 4 levels for entity identification provides enough flexibility
+to better namespace the devices and applications running on it minimizing the risk of `id` conflicts.
+Thin-edge doesn't enforce/dictate how a user uses these namespace levels.
+But there are conventions that can be followed to cover a wide range of use-cases, as follows:
+
+* `te/<device-namespace>/<device-id>/<service-namespace>/<service-id>/...`
+* `te/<device-namespace>/<device-id>/<device-local-namespace>/<device-local-component-id>/...`
+* `te/<entity-namespace>/<entity-id>/<component-namespace>/<component-id>/...`
+* `te/<device-family>/<device-series>/<part-series>/<part-id>/...`
+
+Even for the data type levels, a user is free to define those as they wish.
+But thin-edge has some pre-defined `<data-type>` subtopics for well-known types like:
+`measurements`, `alarms`, `events` and `commands`, on which it enforces some constraints.
```

Comment:
I would not stress that the user is free to define other topic names. He can always do to that. The point is what support will be provided by thin-edge then?

Reply:
The short forms `m` (measurement), `e` (events) were chosen for telemetry data are likely to be published at high frequency. This reduces bandwidth. Other data which is not sent at such high frequencies, e.g. commands can use multiple characters, e.g. `cmd`.